test/NetEscapades.AspNetCore.SecurityHeaders.Test/SecurityHeadersMiddlewareTests.cs

```diff
@@ -1129,6 +1129,544 @@ public async Task SecureRequest_WithExpectCt_WhenUsingTheManualValues_IsCorrect(
+                                builder.AddReport().To("coop_endpoint");
+                            }, asReportOnly: false));
+                    app.Run(async context =>
+                    {
+                        context.Response.ContentType = "text/html";
+                        await context.Response.WriteAsync("Test response");
+                    });
+                });
+
+            using (var server = new TestServer(hostBuilder))
+            {
+                // Act
+                // Actual request.
+                var response = await server.CreateRequest("/")
+                    .SendAsync("PUT");
+
+                // Assert
+                response.EnsureSuccessStatusCode();
+
+                (await response.Content.ReadAsStringAsync()).Should().Be("Test response");
+                var header = response.Headers.GetValues("Cross-Origin-Opener-Policy").FirstOrDefault();
+                header.Should().NotBeNull();
+                header.Should().Be("unsafe-none; report-to=\"coop_endpoint\"");
+                response.Headers.Contains("Cross-Origin-Opener-Policy-Report-Only").Should().BeFalse();
+            }
+        }
+
+        [Fact]
+        public async Task HttpRequest_WithCrossOriginEmbedderPolicyHeaderReportOnly_SetsUnsafeNone()
+        {
+            // Arrange
+            var hostBuilder = new WebHostBuilder()
+                .Configure(app =>
+                {
+                    app.UseSecurityHeaders(
+                        new HeaderPolicyCollection()
+                            .AddCrossOriginEmbedderPolicyReportOnly(builder =>
+                            {
+                                builder.UnsafeNone();
+                                builder.AddReport().To("coep_endpoint");
+                            }));
+                    app.Run(async context =>
+                    {
+                        context.Response.ContentType = "text/html";
+                        await context.Response.WriteAsync("Test response");
+                    });
+                });
+
+            using (var server = new TestServer(hostBuilder))
+            {
+                // Act
+                // Actual request.
+                var response = await server.CreateRequest("/")
+                    .SendAsync("PUT");
+
+                // Assert
+                response.EnsureSuccessStatusCode();
+
+                (await response.Content.ReadAsStringAsync()).Should().Be("Test response");
+                var header = response.Headers.GetValues("Cross-Origin-Embedder-Policy-Report-Only").FirstOrDefault();
+                header.Should().NotBeNull();
+                header.Should().Be("unsafe-none; report-to=\"coep_endpoint\"");
+                response.Headers.Contains("Cross-Origin-Embedder-Policy").Should().BeFalse();
+            }
+        }
+
+        [Fact]
+        public async Task HttpRequest_WithCrossOriginEmbedderPolicyHeaderReportOnly_UsingBoolean_SetsUnsafeNone()
```

Comment:
Yikes, I think I definitely need to refactor these tests to remove the duplication! Thanks for adding them though 🙂